src/manage_sql.c

```diff
@@ -29483,6 +29483,8 @@ print_report_xml_start (report_t report, report_t delta, task_t task,
              "</delta>");
     }
 
+  count_filtered = (delta == 0 && ignore_pagination && get->details);
```

Comment:
This could perhaps also be used in any case where the number of rows is unlimited via a `rows=-1` in the filter and the "Max Rows" limit is disabled.
If "Max Rows" is enabled it could also be compared against the unfiltered total or some other upper bound that's quick to query to see if using the number of returned results could give the wrong count.

Reply:
Sounds like a good idea.  Would like to leave it for a later PR though.